dev/devicelab/lib/framework/running_processes.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+@immutable
+class RunningProcessInfo {
+  const RunningProcessInfo(this.pid, this.creationDate, this.commandLine)
+      : assert(pid != null),
+        assert(commandLine != null);
+
+  final String commandLine;
+  final int pid;
+  final DateTime creationDate;
+
+  @override
+  bool operator ==(Object other) {
+    return other is RunningProcessInfo && other.pid == pid;
+  }
+
+  @override
+  int get hashCode => pid.hashCode;
```

Comment:
Same as above

Reply:
`x ^ y ^ c` is a bad hashcode but its good enough for this.

Reply:
Went with a marginally better option and a TODO for when Dart lands https://dart-review.googlesource.com/c/sdk/+/73360